kong/db/schema/entities/routes.lua

```diff
@@ -4,15 +4,57 @@ local deprecation = require("kong.deprecation")
 
 local validate_route
 do
+  local tonumber = tonumber
+  local re_match = ngx.re.match
+  local re_gmatch = ngx.re.gmatch
+
   local get_schema = require("kong.router.atc").schema
   local get_expression = require("kong.router.compat").get_expression
   local transform_expression = require("kong.router.expressions").transform_expression
 
+  local HTTP_PATH_SEGMENTS_PREFIX = "http.path.segments."
+  local HTTP_PATH_SEGMENTS_REG = [[http\.path\.segments\.(\w*)]]
+  local HTTP_PATH_SEGMENTS_SUFFIX_REG = [[^(\d+)(_(\d+))?$]]
+
+  local function verify_http_path_segments(exp)
+    if not exp:find(HTTP_PATH_SEGMENTS_PREFIX, 1, true) then
```

Comment:
This doesn't feel quite right. Can we build a temporary `router` and use `r:get_fields()` to get the fields? Use of regex is not only harder, but it could break if user write `http.path == "/http.path.segments.1"`.

Reply:
We will do it when atc-router bumped.

Reply:
done.